### PR TITLE
Show Alert message in Facility Locator when the vets-api returns a non-200 response

### LINKED ISSUE #29076: Display a helpful error message in Facility Locator when error response is received from search API
# Description

As a Veteran, I am informed when my facility search fails due to a back-end error, so that I know I can try my search again later.

# Tasks
- [ ] Get design approval on proposed error message wording and format below.

# Acceptance Criteria

- [ ] When calls to vets-api to fetch facilities return an HTTP response code other than 200, and no facilities are returned, display an error message:

### PR DESCRIPTION
**Pull request by mpelzsherman**
Resolves https://github.com/department-of-veterans-affairs/va.gov-team/issues/29076

Displays an error alert when the Facility Locator receives a bad (non-200) API response.

Note: Use of the `MockLocatorApi ` in e2e tests was preventing us from mocking the error response in Cypress using `cy.intercept`. It also added a bunch of test-specific code, so it prevented the e2e tests from actually testing the "real" API code. So I've deleted it! 

We should be using `mockFetch` in our unit tests as needed, but none of our unit tests actually trigger any fetches, so we're good.

### CODE REVIEW

**Review by ndsprinkle — APPROVED** [review]

**Comment by mpelzsherman** [comment]
Thanks @ndsprinkle but this PR is still in draft. You may want to re-review after I've marked it ready for review.

**Comment by ndsprinkle** [comment]
@mpelzsherman Ah sorry about that, wasn't paying attention. I will be sure to re-review later!

File: src/applications/facility-locator/api/index.js [hunk 1]
```diff
@@ -3,11 +3,7 @@ import LiveApi from './LocatorApi';
 
 const getAPI = () => {
   const isUnitTest = window.Mocha;
-  const isCypress = window.Cypress;
-
-  if (isUnitTest || isCypress) return MockApi;
-
-  return LiveApi;
+  return isUnitTest ? MockApi : LiveApi;
```
Comment (mpelzsherman) [inline src/applications/facility-locator/api/index.js:6]: Use of the `MockLocatorApi ` in e2e tests was preventing us from mocking the error response in Cypress using `cy.intercept`. It also adds a bunch of test-specific code, so it prevents the e2e tests from actually using the "real" API code. We should be using `mockFetch` in our unit tests as needed, and get rid of trhe `MockLocatorApi` altogether ASAP.

**Review by kelsonic — APPROVED** [review]
lgtm!

**Review by ndsprinkle — APPROVED** [review]

**Pull request merged**